.ci/install_kata_kernel.sh

```diff
@@ -13,80 +13,75 @@ set -e
 cidir=$(dirname "$0")
 source "${cidir}/lib.sh"
 
-repo_owner="clearcontainers"
-repo_name="linux"
-
-linux_releases_url="https://github.com/${repo_owner}/${repo_name}/releases"
-#fake repository dir to query kernel version from remote
-fake_repo_dir=$(mktemp -t -d kata-kernel.XXXX)
-
-function cleanup {
-	rm  -rf "${fake_repo_dir}"
+repo_name="packaging"
+repo_owner="kata-containers"
+kata_kernel_dir="/usr/share/kata-containers"
+kernel_arch="$(arch)"
+get_kernel_url="https://cdn.kernel.org/pub/linux/kernel"
+tmp_dir=$(mktemp -d)
+hypervisor="kvm"
+
+download_repo() {
+	pushd ${tmp_dir}
+	git clone --depth 1 https://github.com/${repo_owner}/${repo_name}
+	popd
 }
 
-trap cleanup EXIT
-
-function usage() {
-	cat << EOT
-Usage: $0 <version>
-Install the containers clear kernel image <version> from "${repo_owner}/${repo_name}".
-
-version: Use 'latest' to pull latest kernel or a version from "${cc_linux_releases_url}"
-EOT
-
-	exit 1
+get_kernel_version() {
+	pushd $tmp_dir/$repo_name > /dev/null
+	kernel_version=$(grep "Linux/[${kernel_arch}]*" kernel/configs/[${kernel_arch}]*_kata_${hypervisor}_* | cut -d' ' -f3 | tail -1)
+	popd > /dev/null
+	if [ -z "$kernel_version" ]; then
+		die "unknown kernel version for architecture $kernel_arch"
+	else
+		echo ${kernel_version}
+	fi
 }
 
-#Get latest version by checking remote tags
-#We dont ask to github api directly because force a user to provide a GITHUB token
-function get_latest_version {
-	pushd "${fake_repo_dir}" >> /dev/null
-	git init -q
-	git remote add origin  https://github.com/"${repo_owner}/${repo_name}".git
-
-	cc_release=$(git ls-remote --tags 2>/dev/null \
-		| grep -oP '\-\d+\.container'  \
-			| grep -oP '\d+' \
-				| sort -n | \
-					tail -1 )
-
-	tag=$(git ls-remote --tags 2>/dev/null \
-		| grep -oP "v\d+\.\d+\.\d+\-${cc_release}.container" \
-			| tail -1)
-
-	popd >> /dev/null
-	echo "${tag}"
+# download the linux kernel, first argument is the kernel version
+download_kernel() {
+	kernel_version=$1
+	pushd $tmp_dir
+	kernel_tar_file="linux-${kernel_version}.tar.xz"
+	kernel_url="${get_kernel_url}/v$(echo $kernel_version | cut -f1 -d.).x/${kernel_tar_file}"
```

Comment:
I prefer the version.yaml is got from versions.yaml in runtime.

Reply:
qq, is there a parser in bash for yaml?

Reply:
also If I update the kernel version I need to update the yaml file in a different PR at the same time